### PR TITLE
Adding user profile fields

### PR DESCRIPTION
**Pull request by braydonwang**
## Notion Ticket Link

[Add User Profile Fields](https://www.notion.so/uwblueprintexecs/Add-User-Profile-Fields-d7c41160b54641018752b4c1ad394f94#2fa82d9737fe451092a87bb492fc515f)

## Implementation Description
* Added additional fields to user schema
* Created custom scalar Date type with its resolver

## Steps to Test
1. Nothing at the moment, will need to wait for api endpoints

## What should reviewers focus on?
* Look over the resolver and make sure it's right, I followed this doc: https://www.apollographql.com/docs/apollo-server/schema/custom-scalars/

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### Mac
```bash
docker exec -it CWC-backend /bin/bash -c "yarn fix"
docker exec -it CWC-frontend /bin/bash -c "yarn fix"
```

### Windows
```bash
docker exec -it CWC-backend bash -c "yarn fix"
docker exec -it CWC-frontend bash -c "yarn fix"
```

### CODE REVIEW

File: backend/models/user.model.ts [hunk 1]
```diff
@@ -28,6 +32,23 @@ const UserSchema: Schema = new Schema({
...
+    required: true,
+  },
+  birthday: {
+    type: Date,
+    required: true,
+  },
+  locationType: {
+    type: String,
+    required: true,
+    enum: ["Remote", "Hybrid", "In Person"],
```
Comment (shencynthia) [inline backend/models/user.model.ts:46]: In the types file, for the Location Type enum field you have "InPerson" (no space) v.s "In Person" here.

Comment (shencynthia) [inline backend/models/user.model.ts:46]: In the enum in userType you have "InPerson" and here its "In Person".

File: backend/types.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 export type Role = "Volunteer" | "Staff" | "Admin";
 
+export type LocationType = "Remote" | "Hybrid" | "In Person";
```
Comment (shencynthia) [inline backend/types.ts:3]: Same thing here, "InPerson" v.s "In Person"

File: backend/types.ts [hunk 3]
```diff
@@ -11,6 +13,10 @@ export type UserDTO = {
   lastName: string;
   email: string;
   role: Role;
+  startDate?: Date;
+  birthday?: Date;
+  locationType?: LocationType;
+  headshot?: string;
```
Comment (shencynthia) [inline backend/types.ts:19]: In our Data Model we have headshot as non-optional.

**Review by shencynthia — CHANGES_REQUESTED** [review]
Just some small changes for consistency and also a nit: use imperative tense for commits + PR (e.g. Add rather than Adding).

Otherwise, looks good!!

**Review by shencynthia — APPROVED** [review]
LGTM! 👍👍👍

File: backend/models/user.model.ts [hunk 4]
```diff
@@ -28,6 +32,23 @@ const UserSchema: Schema = new Schema({
...
+  },
+  locationType: {
+    type: String,
+    required: true,
+    enum: ["Remote", "Hybrid", "InPerson"],
+  },
+  headshot: {
+    type: String,
+    required: true,
+  },
```
Comment (carolynzhang18) [inline backend/models/user.model.ts:51]: Same comment as before here: these probably shouldn't be `required` - the current signup functionality will break :(

**Review by owen-sellner — APPROVED** [review]

**Pull request merged**